### PR TITLE
Add LogKey to plugins dir so that it can be imported by all go plugins

### PR DESCRIPTION
**Pull request by PettitWesley**
This code is directly copied from the CloudWatch plugin repo; the plugins dir in this repo is used for common code which is imported in all plugins.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by sonofachamp** [comment]
Is there any intent to utilize this `LogKey` function in other plugins, or is it specific to the CloudWatch plugin? I'd suggest we stick to the "a little copying is better than a little dependency" strategy until we have enough duplication to justify a common module.

**Comment by PettitWesley** [comment]
@sonofachamp Yes, the plan is to use this function in all 3 of the plugins

File: plugins/plugins.go [hunk 1]
```diff
@@ -62,6 +63,29 @@ func (t *Timeout) Check() {
 	}
 }
 
+// Implements the log_key option, which allows customers to only send the value of a given key to CW Logs
```
Comment (sonofachamp) [inline plugins/plugins.go]:
```suggestion
// LogKey returns the value associated with the input key from the record map, or an error if the key is not found.
```

**Review by sonofachamp — APPROVED** [review]
Minor change suggested in the doc comment.

**Pull request merged**